modules/argocd/main.tf

```
@@ -96,6 +96,46 @@ resource "helm_release" "argocd_application" {
     })
   ]
 
+  depends_on = [module.helm_addon, helm_release.argocd_project]
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# ArgoCD Projects Bootstrapping
+# ---------------------------------------------------------------------------------------------------------------------
+resource "helm_release" "argocd_project" {
+  for_each = { for k, v in var.projects : k => merge(local.default_argocd_project, v) }
+
+  name      = each.key
+  chart     = "${path.module}/argocd-project/helm"
+  version   = "1.0.0"
+  namespace = local.helm_config["namespace"]
```

```suggestion
  namespace = lookup(each.value, "namespace", [])
```
allow to override the namespace from the map `projects`

---

updated but it is still using `namespace = local.helm_config["namespace"]` in `applications`.

---

Yes this is intentional, we want to allow the flexibility to install application and projects in different namespaces, and this way if user doesn't specify a different namespace via project.namepspace it defaults to `local.helm_config["namespace"]` via:
```
default_argocd_project = {
    namespace   = local.helm_config["namespace"]
```